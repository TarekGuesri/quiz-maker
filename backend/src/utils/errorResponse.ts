export class ErrorResponse extends Error {
  statusCode = 0;
  constructor(message: string, statusCode: number) {
    super(message);
    this.statusCode = statusCode;

    Error.captureStackTrace(this, this.constructor);
  }
}
