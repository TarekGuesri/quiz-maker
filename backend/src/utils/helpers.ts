export const trimmer = (text: string): string => {
  if (text) {
    return text.trim();
  } else {
    return text;
  }
};
