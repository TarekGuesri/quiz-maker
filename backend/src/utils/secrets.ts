import dotenv from "dotenv";
import fs from "fs";

if (fs.existsSync(".env")) {
  console.log("Using .env file to supply config environment variables");
  dotenv.config({ path: ".env" });
} else {
  console.log("Using .env file to supply config environment variables");
  dotenv.config({ path: ".env.example" }); // you can delete this after you create your own .env file!
}

export const ENVIRONMENT = process.env.NODE_ENV;
const prod = ENVIRONMENT === "production"; // Anything else is treated as 'dev'

export const MONGODB_URI = prod
  ? process.env["MONGODB_URI"]
  : process.env["MONGODB_URI_LOCAL"];

if (!MONGODB_URI) {
  if (prod) {
    console.error(
      "No mongo connection string. Set MONGODB_URI environment variable.",
    );
  } else {
    console.error(
      "No mongo connection string. Set MONGODB_URI_LOCAL environment variable.",
    );
  }
  process.exit(1);
}
