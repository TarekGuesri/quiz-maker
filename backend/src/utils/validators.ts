import { Schema } from "express-validator";

export const quizSchema: Schema = {
  title: {
    // isLength: { errorMessage: "Title can't be empty", options: { min: 1 } },
    errorMessage: "Title can't be empty!",
    notEmpty: true,
  },
  questions: {
    isArray: { options: { min: 1, max: 10 } },
    errorMessage: "Questions must be an array of a length between 1 and 10!",
  },
  "questions.*.content": {
    errorMessage: "Question content can't be empty!",
    notEmpty: true,
  },
};
