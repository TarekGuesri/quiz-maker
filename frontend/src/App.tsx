import React from 'react';
import Button from '@mui/material/Button';
import './App.css';

const App: React.FC = () => {
  return (
    <div className="App">
      <Button variant="contained">Hello World</Button>
    </div>
  );
};

export default App;
