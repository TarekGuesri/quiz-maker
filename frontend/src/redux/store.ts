import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import ui from "./ui/ui-slice";

export const store = configureStore({
  reducer: {
    ui,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
