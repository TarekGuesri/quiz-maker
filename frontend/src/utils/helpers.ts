export const timeFormatter = (
  seconds: number,
  minutes: number,
  hours: number,
): string => {
  const secondsString = seconds.toString().padStart(2, "0");
  const minutesString = minutes.toString().padStart(2, "0");

  if (hours === 0) {
    return `${minutesString}:${secondsString}`;
  } else {
    const hoursString = hours.toString().padStart(2, "0");

    return `${hoursString}:${minutesString}:${secondsString}`;
  }
};
