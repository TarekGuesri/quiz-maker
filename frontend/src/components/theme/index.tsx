import React, { FC } from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import PropTypes from 'prop-types';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { useAppSelector } from 'src/redux/hooks';
import varelaRound from 'src/assets/fonts/VarelaRound-Regular.ttf';

export const Theme: FC = (props) => {
  const { darkMode } = useAppSelector((state) => state.ui);

  const theme = createTheme({
    palette: {
      mode: darkMode ? 'dark' : 'light',
      background: {
        default: darkMode ? '#333' : '#f7f7f7',
        paper: darkMode ? '#333' : '#fbfbfb',
      },
      primary: {
        main: '#8e49de',
        // contrastText:
      },
      secondary: {
        main: '#f9de4a',
      },
      contrastThreshold: 3,
      tonalOffset: 0.2,
    },
    typography: {
      fontFamily: 'Varela Round',
    },
    components: {
      MuiCssBaseline: {
        styleOverrides: `
          @font-face {
            font-family: 'Varela Round';
            font-style: normal;
            font-display: swap;
            font-weight: 400;
            src: url(${varelaRound}) format('truetype');
        `,
      },
    },
  });

  return (
    <ThemeProvider theme={theme}>
      {props.children}
      <CssBaseline />
    </ThemeProvider>
  );
};

Theme.propTypes = {
  children: PropTypes.node.isRequired,
};
