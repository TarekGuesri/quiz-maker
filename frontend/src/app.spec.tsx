/**
 * @jest-environment jsdom
 */

import React from "react";
import { cleanup, render } from "@testing-library/react";
import App from "./app";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import { Theme } from "src/components/theme";
import { store } from "src/redux/store";
import { history } from "src/utils/history";

describe("src/app.tsx", () => {
  afterEach(() => cleanup());
  test("Render App component", () => {
    const renderedApp = render(
      <Provider store={store}>
        <Router history={history}>
          <Theme>
            <App />
          </Theme>
        </Router>
      </Provider>,
    );
    expect(renderedApp.container).toMatchSnapshot();
  });
});
