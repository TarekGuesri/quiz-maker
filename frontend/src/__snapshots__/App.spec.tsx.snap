// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/entry/app.tsx Render App component 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1rcbgiz"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-c0kq1a-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular makeStyles-toolbar-1 css-1t29gy6-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h4 css-eydqpw-MuiTypography-root"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
              style="text-decoration: none;"
            >
              QuizMaker
            </a>
          </div>
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-x7opdv-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-jqqqr0-MuiTypography-root"
              >
                ðŸŒ™
              </span>
            </label>
          </div>
        </div>
      </header>
    </div>
    <div
      class="makeStyles-box-3 MuiBox-root css-0"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary makeStyles-root-2 css-rmd5o9-MuiCircularProgress-root"
        role="progressbar"
        style="width: 50px; height: 50px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiBox-root css-1swgbew"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 css-2jfkze-MuiTypography-root"
      >
        Â© 
        2022
         TARIQ LAMIN GUESRI. ALL RIGHTS RESERVED.
      </div>
    </div>
  </div>
</div>
`;
